bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/LspPlugin.java

```diff
@@ -50,6 +53,9 @@ public void start(BundleContext context) throws Exception {
 		workspaceTracker.open();
 		workspace = workspaceTracker.getService();
 		cLanguageServerProvider = new CLanguageServerRegistry().createCLanguageServerProvider();
+		
+		// Disable warnings, see https://github.com/eclipse-cdt/cdt-lsp/issues/88
+		Logger.getLogger("").setLevel(Level.SEVERE);
```

Comment:
With the fix in https://github.com/eclipse/tm4e/pull/535/files does that mean this line isn't needed? It seems quite overkill for CDT LSP to set the level of the logger here.

Reply:
> With the fix in https://github.com/eclipse/tm4e/pull/535/files does that mean this line isn't needed? It seems quite overkill for CDT LSP to set the level of the logger here.

Unfortunately not. The tm4e fix changed that instead of always printing to `stderr`, the logger is used (warning level). However, we could also just set the logging level for the class which is concerned. I didn't do this in this PR, because it would somehow introduce a hidden dependency (e.g. if tm4e renames the class, the log messages will reappear). 

So I think what should also work is (I didn't test this):
```
Logger.getLogger("org.eclipse.tm4e.core.internal.oniguruma.OnigRegExp").setLevel(Level.SEVERE);
```

Reply:
I think for now we should do `Logger.getLogger("org.eclipse.tm4e.core.internal.oniguruma.OnigRegExp").setLevel(Level.SEVERE);` as not to affect the logging level of the whole system as there are lots of other pieces in play. 

I have another task of reviewing logging (in context of https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/588#issuecomment-1565332227 / https://github.com/eclipse-packaging/packages/issues/27)

Reply:
Raised #101

Reply:
I agree, I created a PR: #102